crates/context/config/src/client.rs

```diff
@@ -63,199 +64,237 @@ impl Client<AnyTransport> {
         });
 
         let local = Self::from_local_config(&config).expect("validation error");
-
         let transport = transport!(local.transport, relayer);
-
         Self::new(transport)
     }
 
     pub fn from_local_config(config: &ClientConfig) -> eyre::Result<Client<LocalTransports>> {
         let mut near_transport = None;
+        let mut starknet_transport = None;
+        let mut ethereum_transport = None;
+        let mut icp_transport = None;
+        let mut stellar_transport = None;
 
-        'skipped: {
-            if let Some(near_config) = config.signer.local.protocols.get("near") {
-                let Some(e) = config.params.get("near") else {
-                    eyre::bail!("missing config specification for `{}` signer", "near");
-                };
-
-                if !matches!(e.signer, ClientSelectedSigner::Local) {
-                    break 'skipped;
-                }
+        'near: {
+            let Some(params) = config.params.get("near") else {
+                break 'near;
+            };
 
-                let mut config = near::NearConfig {
-                    networks: Default::default(),
-                };
+            // Strict local signer check
+            if !matches!(params.signer, ClientSelectedSigner::Local) {
+                break 'near;
+            }
 
-                for (network, signer) in &near_config.signers {
-                    let Credentials::Near(credentials) = &signer.credentials else {
-                        eyre::bail!("expected Near credentials but got {:?}", signer.credentials)
-                    };
-
-                    let _ignored = config.networks.insert(
-                        network.clone().into(),
-                        near::NetworkConfig {
-                            rpc_url: signer.rpc_url.clone(),
-                            account_id: credentials.account_id.clone(),
-                            access_key: credentials.secret_key.clone(),
-                        },
-                    );
-                }
-
-                near_transport = Some(near::NearTransport::new(&config));
+            let near_config = config.signer.local.protocols.get("near").ok_or_else(|| {
+                eyre::eyre!("Local signer selected for Near but no configuration found")
+            })?;
+
+            let mut network_config = near::NearConfig {
+                networks: BTreeMap::new(),
+            };
+
+            for (network, signer) in &near_config.signers {
+                // Direct credential access
+                let credentials = signer.near_credentials.as_ref().ok_or_else(|| {
+                    eyre::eyre!("Near credentials missing for network {}", network)
+                })?;
```

Comment:
this should be `serde_json::from_value` where you deserialize to `near::Credentials`